### ISSUE TITLE
suggestions for dimensions

### ISSUE THREAD

**Issue by mirlip** [mauge123/mechanical-blender#3]
Hi,
First of all, awesome job guys! The last fix worked like a charm, thank you. The new non-modal transform is really much better than previously. The dimensions are really well designed and pretty intuitive to use. But it is hard to read the text sometime. Would it be possible to:
- have the possibility to display those dimensions in object mode
- allow to customize the font (add it's own category into the theme to change the size, color) would really help. 
- have the text more outside or to make a hole in the line where the number is displayed to make it more readable.
Regards

**Comment by mauge123** [comment]
Thanks for your comments ;)

-  It's on plan, to have a section on 3D view properties with a flag to show all dimensions on object mode,or only the selected object.

- It's a good a place to set all this customizations.

- This will be settings that will be set in same place that 2)

PD: note that dimension information is saved on .blend file with mesh data. This means that if the file is opened and saved with a different binary (not from this project) dimension data will be loose. Also be careful as it's on development, and should not be considered stable, but is good to have people testing/using it ;)

**Comment by mirlip** [comment]
Thank you for your quick answer ;) Can't wait :)

**Comment by mauge123** [comment]
56fab086d4b4c50362c7a94ccea39970ff4d3057  includes the first point, enabling or disabling mesh_dismensions in object mode.

**Comment by mirlip** [comment]
Hi Mauge,
Nice :) Compiling on windows gives following errors: blender\source\blender\modifiers\intern\MOD_fluidsim_util.c(218): error C2198: 'CDDM_new' : too few arguments for call
Note that it does this error for all modifiers and for smoke.c

**Comment by mauge123** [comment]
72fc37c22d55f403c7c02de414782070d790bf32 should ammend it.

I've have not detected because i'm working a really basic setting (all possible is disabled). I've changed the way it was done, as the function may be used around, and can due to future crashes.

Thanks!

**Comment by mauge123** [comment]
277f5f6dbf1f4585eab1deded1e34c7dc465c01a

I'll try to set a more complete setting for testings this issues.   

Thanks!

**Comment by mirlip** [comment]
VS2013 still fails with following error:
```
Creating library C:/Users/test/git/build_windows_full/bin/Release/blenderplayer.lib and object C:/Users/test/git/build_windows_full/bin/Release/blenderplayer.exp
bf_bmesh.lib(bmesh_delete.obj) : error LNK2019: unresolved external symbol get_selected_dimension_BMesh referenced in function BMO_mesh_delete_oflag_context [C:\Users\test\git\build_windows_full\source\blenderplayer\blenderplayer.vcxproj]
C:\Users\test\git\build_windows_full\bin\Release\blenderplayer.exe : fatal error LNK1120: 1 unresolved externals 
```

**Comment by mirlip** [comment]
It compiles without player, and show dimensions work. But show dimensions on selected doesn't work. I also tried to duplicate (shift-D) a cube with dimentions (visible in object mode), entered edit mode on this copy, modified the dimension, then hit tab to go back to object mode and it crashed.

**Comment by mauge123** [comment]
Show dimensions on selected works for me, but maybe is unclear, as show dimensions must be selected also.

Commited a fix solving crash related to duplicate.

Thanks!

**Comment by mirlip** [comment]
Thanks for the fix. For the selection only view, maybe call it restrict to selected object and make it greyed out when view selected is not activated?

**Comment by mirlip** [comment]
Just tested the fix, works great :) Really good job. Slowly playing with it in real cases (not saving files as they may break, please feel free to change everything that is needed ;). It may help readability to have an option to limit the display of dimensions to the dimensions being in the current view plane: only dimensions in the x,y plane when in top or bottom view, etc.... Maybe a threshold would help for dimensions that are not perfectly coplanar, but should still be displayed?

**Comment by mauge123** [comment]
41964561efe70ff37a5b1a864dc529d8cc20019c enables color customization fixed on 1a71a2ead220c7ab0df10ed502ef3a647b9e5638 (with WITH_PLAYER enabled)
